sdk/golang/ccnp/quote/quote.go

```diff
@@ -0,0 +1,191 @@
+	Reserved     uint32    // Reserved field.
+	Vendor_id    [16]uint8 // Unique identifier of QE Vendor.
+	User_data    [20]uint8 // Custom attestation key owner data.
+}
+
+type TDReport struct {
+	Tee_tcb_svn     [16]uint8
+	Mrseam          [48]uint8
+	Mrsignerseam    [48]uint8
+	Seam_attributes [8]uint8
+	Td_attributes   [8]uint8
+	Xfam            [8]uint8
+	Mrtd            [48]uint8
+	Mrconfigid      [48]uint8
+	Mrowner         [48]uint8
+	Mrownerconfig   [48]uint8
+	Rtmrs           [192]uint8
+	Report_data     [64]uint8
+}
+
+const (
+	Quote_Header_Index                    = 0   // 48 bytes quote header, start from index 0 of quote string
+	Quote_TDReport_Index                  = 48  // 584 bytes tdreport, start from index 48 of quote string
+	Quote_Auth_data_size_Index            = 632 // 4 bytes auth size, start from index 632 of quote string
+	Quote_Auth_data_content_Index         = 636 // auth_size bytes in auth_data, start from index 636 of quote string
+	Quote_Auth_data_signature_Index       = 700 // 64 bytes of signature in auth_data, start from index 700 of quote string
+	Quote_Auth_data_attestation_key_Index = 764 // 64 bytes of attestation_key in auth_data, start from index 764 of quote string
+	Quote_Auth_data_cert_data_Index       = 770 // (auth_size-6-128) bytes of cert_data in auth_data, start from index 770 of quote string
+)
+
+func GetQuote(user_data string, nonce string) (interface{}, error) {
+
+	channel, err := grpc.Dial(UDS_PATH, grpc.WithInsecure())
+	if err != nil {
+		log.Fatalf("[GetQuote] can not connect to UDS: %v", err)
+	}
+	defer channel.Close()
+
+	client := pb.NewGetQuoteClient(channel)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	response, err := client.GetQuote(ctx, &pb.GetQuoteRequest{UserData: user_data, Nonce: nonce})
+	if err != nil {
+		log.Fatalf("[GetQuote] fail to get quote: %v", err)
+	}
+
+	quote, err := base64.StdEncoding.DecodeString(strings.Trim(response.Quote, "\""))
+	if err != nil {
+		log.Fatalf("[GetQuote] decode quote error: %v", err)
+	}
+
+	switch response.QuoteType {
+	case TYPE_TDX:
+		return parseTDXQuote(quote)
+	case TYPE_TPM:
+		return parseTPMQuote(quote)
+	default:
+		log.Fatalf("[GetQuote] unknown TEE enviroment!")
+	}
+
+	return nil, pkgerrors.New("[GetQuote] unknown TEE enviroment!")
+}
+
+func parseTDXQuote(quote []byte) (interface{}, error) {
+
+	var header = SGX_Quote_Header{}
+	var err = binary.Read(bytes.NewReader(quote[Quote_Header_Index:Quote_TDReport_Index]), binary.LittleEndian, &header)
+	if err != nil {
+		log.Fatalf("[parseTDXQuote] fail to parse quote header: %v", err)
+	}
+
+	var tdreport = TDReport{}
+	err = binary.Read(bytes.NewReader(quote[Quote_TDReport_Index:Quote_Auth_data_size_Index]), binary.LittleEndian, &tdreport)
+	if err != nil {
+		log.Fatalf("[parseTDXQuote] fail to parse quote tdreport: %v", err)
```

Comment:
shall we return instead of moving on under such Fatal errors scenario?

Reply:
Fatalf will exit